### PR TITLE
Update SDK version to 3.0.0

### PR DESCRIPTION
**Pull request by prm-james-hill**

### CODE REVIEW

File: src/main/java/ai/promoted/example/App.java [hunk 1]
```diff
@@ -61,6 +51,10 @@ public class App implements Callable<Integer>
         defaultValue = "false")
     private boolean blockingShadowTraffic = false;
 
+    @Option(names = "--useGrpc", description = "Whether to use gRPC for delivery calls.",
+        defaultValue = "false")
+    private boolean useGrpc;
```
Comment (prm-dan) [inline src/main/java/ai/promoted/example/App.java:56]: I'm fine if you want to default to true.

File: src/main/java/ai/promoted/example/App.java [hunk 2]
```diff
@@ -83,38 +77,35 @@ public Integer call() throws Exception {
...
-        UserInfo userInfo = new UserInfo();
-        userInfo.setAnonUserId("anonUserId1");
-        request.setUserInfo(userInfo);
-        request.setUseCase(UseCase.SEARCH);
-        request.setSearchQuery("query");
-        request.setPaging(new Paging().offset(0).size(2));
-        request.setDisablePersonalization(false); // default=false.
+    private static Request.Builder newTestRequestBuilder() {
+        Request.Builder requestBuilder = Request.newBuilder();
+        UserInfo.Builder userInfoBuilder = UserInfo.newBuilder();
```
Comment (prm-dan) [inline src/main/java/ai/promoted/example/App.java:94]: JFYI - you can also do 
```
UserInfo.Builder userInfoBuilder = request.getUserInfoBuilder()
userInfoBuilder.setAnonUserId("anonUserId1");
```
Then you don't need the extra line to `setUserInfo`.

File: src/main/java/ai/promoted/example/App.java [hunk 3]
```diff
@@ -83,38 +77,35 @@ public Integer call() throws Exception {
...
+        requestBuilder.setUseCase(UseCase.SEARCH);
+        requestBuilder.setSearchQuery("query");
+        requestBuilder.setPaging(Paging.newBuilder().setOffset(0).setSize(2));
+        requestBuilder.setDisablePersonalization(false); // default=false.
         for (int i = 0; i < 3; i++) {
-            Insertion insertion = new Insertion();
-            insertion.setContentId("content" + i);
-            insertion.setRetrievalRank(i);
-            request.addInsertionItem(insertion);
+            Insertion.Builder insertionBuilder = Insertion.newBuilder();
```
Comment (prm-dan) [inline src/main/java/ai/promoted/example/App.java:102]: Same.  I think it's `requestBuilder.addInsertionBuilder()`

**Review by prm-dan — APPROVED** [review]

File: src/main/java/ai/promoted/example/App.java (hunk 1 above)
Comment (prm-james-hill) [reply src/main/java/ai/promoted/example/App.java:56]: Done

File: src/main/java/ai/promoted/example/App.java (hunk 2 above)
Comment (prm-james-hill) [reply src/main/java/ai/promoted/example/App.java]: Done

File: src/main/java/ai/promoted/example/App.java (hunk 3 above)
Comment (prm-james-hill) [reply src/main/java/ai/promoted/example/App.java]: Done

**Pull request merged**